inc/compatibility/fixes/class-acceleratedmobilepagesfix.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Accelerated Mobile Pages Fix
+ *
+ * @package Pantheon\Compatibility\Fixes
+ */
+
+namespace Pantheon\Compatibility\Fixes;
+
+/**
+ * Accelerated Mobile Pages Fix
+ *
+ * @package Pantheon\Compatibility\Fixes
+ */
+class AcceleratedMobilePagesFix {
+
+
+
+	public static function apply() {
+		SelfUpdatingThemesFix::apply();
+		global $redux_builder_amp;
+		// Force disabling AMP mobile redirection.
+		$redux_builder_amp['amp-mobile-redirection'] = 0;
+
+		if ( ( self::is_mobile() ) && ( false === strrpos( $_SERVER['REQUEST_URI'], 'amp' ) ) ) {
+			header( 'HTTP/1.0 301 Moved Permanently' );
+			header( 'Location: https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'] . '/amp' );
```

Comment:
This line is triggering a Guardrails warning: https://github.com/pantheon-systems/pantheon-mu-plugin/pull/48#issuecomment-2141053653

While we largely don't use Guardrails anymore, the point it's making still stands. I'm also concerned about this because there are many instances where `$_SERVER` is not defined (e.g. in CLI running outside the appserver), although this would actually be handled by the `is_mobile` check. There should at least be some validation to ensure that `$_SERVER['HTTP_HOST']` and `$_SERVER['REQUEST_URI']` are set before using them. Or validate and store in a variable to (maybe) make Guardrails happier.